### PR TITLE
fix: upgrade docs readme

### PR DESCRIPTION
**Pull request by erickzhao**
With https://github.com/electron/electron/pull/26335, we updated the hardcoded structure in our docs README. This broke the page layout!

This PR re-organizes the index's content to fit the new structure.

### CODE REVIEW

**Review by bandantonio — APPROVED** [review]

File: views/docs/index.hbs [hunk 1]
```diff
@@ -7,108 +7,109 @@
...
-            <h4 class="docs-sub-heading pb-3 mb-3">
-              {{localized.docs.both_proc_modules}}
-            </h4>
-            <div class="docs-guides-readme">
-              {{{bothProcModules.html}}}
-            </div>
+          <!-- Renderer Process Modules -->
+          <h4 class="docs-sub-heading pb-3 mb-3">
+            {{localized.docs.renderer_proc_modules}}
+          </h4>
```
Comment (malept) [inline views/docs/index.hbs:68]: Should this be an h3 to match the "main process modules" & the "both process types modules" headers?
```suggestion
          <h3 class="docs-sub-heading pb-3 mb-3">
            {{localized.docs.renderer_proc_modules}}
          </h3>
```

**Review by malept — COMMENTED** [review]
I took a quick scan of the preview website's `/docs` and it looks better, I just have one question.

File: views/docs/index.hbs (hunk 1 above)
Comment (erickzhao) [reply views/docs/index.hbs:68]: Oh yeah, good catch. I originally changed those to `h3` for the whole to be more semantic (i.e. not skipping a level of headings), but this one just slipped through.

**Review by malept — APPROVED** [review]
:shipit: after the merge conflicts get fixed

**Pull request merged**